source/rules/no-finnish.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { Rule } from "eslint";
+import * as es from "estree";
+import { couldBeType } from "tsutils-etc";
+import { getTypeCheckerAndNodeMap } from "../utils";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Disallows the use of Finnish notation.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Finnish notation is forbidden"
+    },
+    schema: []
+  },
+  create: context => {
+    const { nodeMap, typeChecker } = getTypeCheckerAndNodeMap(context);
+
+    function report(node: es.Node) {
+      const tsNode = nodeMap.get(node);
+      const tsType = typeChecker.getTypeAtLocation(tsNode);
+
+      if (couldBeType(tsType, "Observable")) {
```

Comment:
Fyi, https://github.com/cartant/eslint-plugin-rxjs/pull/20 added some helper function - it allows you to do `couldBeObservable(node)`